src/storage/txn/scheduler.rs

```diff
@@ -251,6 +269,40 @@ pub struct Scheduler {
     running_write_count: usize,
 }
 
+// Make clippy happy.
+type MultipleReturnValue = (Option<MvccLock>, Vec<(u64, Write)>, Vec<(u64, bool, Value)>);
+
+fn find_mvcc_infos_by_key(reader: &mut MvccReader,
+                          key: &Key,
+                          mut ts: u64)
+                          -> Result<MultipleReturnValue> {
+    let mut writes = vec![];
+    let mut values = vec![];
+    let lock = try!(reader.load_lock(key));
+    loop {
+        let opt = try!(reader.seek_write(key, ts));
+        match opt {
+            Some((commit_ts, write)) => {
+                ts = commit_ts - 1;
+                let write_type = write.write_type;
+                writes.push((commit_ts, write));
+                if write_type != WriteType::Put {
+                    continue;
+                }
+            }
+            None => break,
+        };
+        let write = &writes[writes.len() - 1].1;
+        if let Some(v) = write.short_value.clone() {
+            values.push((write.start_ts, true, v));
+            continue;
+        }
+        let v = try!(reader.load_data(key, write.start_ts));
```

Comment:
Maybe we should scan default cf like `seek_write` too, because data in default cf may not match data in write cf if we have some bugs.